evap/evaluation/tests/test_views.py

```diff
@@ -246,3 +248,44 @@ def test_notebook(self):
 
         user.refresh_from_db()
         self.assertEqual(user.notes, self.note)
+
+
+class TestResetEvaluation(WebTestStaffMode):
+
+    @classmethod
+    def setUpTestData(cls) -> None:
+        cls.manager = make_manager()
+        cls.semester = baker.make(Semester, results_are_archived=True)
+
+    def reset_from_x_to_new(self, x, success_expected: bool) -> None:
+        evaluation = baker.make(Evaluation, state=x, course__semester=self.semester)
+
+        semester_overview_page = self.app.get(
+            reverse("staff:semester_view", args=[self.semester.pk]), user=self.manager, status=200
+        )
+
+        form = semester_overview_page.forms["evaluation_operation_form"]
+
+        form["evaluation"] = [evaluation.pk]
+
+        confirmation_page = form.submit(name="target_state", value=str(Evaluation.State.NEW))
+
+        if success_expected:
+            confirmation_page.forms["evaluation-operation-form"].submit()
+            self.assertEqual(Evaluation.objects.get(pk=evaluation.pk).state, Evaluation.State.NEW)
+        else:
+            self.assertNotIn("evaluation-operation-form", confirmation_page.forms)
+
+    def test_reset_to_new(self) -> None:
+        invalid_start_states = [Evaluation.State.NEW, Evaluation.State.PUBLISHED]
+
+        valid_start_states = [
+            Evaluation.State.PREPARED,
+            Evaluation.State.EDITOR_APPROVED,
+            Evaluation.State.APPROVED,
+        ]
+
+        for s in valid_start_states:
+            self.reset_from_x_to_new(s, True)
+        for s in invalid_start_states:
+            self.reset_from_x_to_new(s, False)
```

Comment:
I'd add the parameter names here to make reading easier:

```suggestion
        for s in valid_start_states:
            self.reset_from_x_to_new(s, success_expected=True)
        for s in invalid_start_states:
            self.reset_from_x_to_new(s, success_expected=False)
```